pkg/controller/nebulabackup/nebula_backup_manager.go

```diff
@@ -71,7 +72,15 @@ func (bm *backupManager) Create(backup *v1alpha1.NebulaBackup) error {
 		return fmt.Errorf("nebula cluster %s/%s is not ready", ns, backup.Spec.BR.ClusterName)
 	}
 
-	backupJob := generateBackupJob(backup, nc.GetMetadThriftConnAddress())
+	if backup.Spec.MaxReservedTime != nil || backup.Spec.MaxBackups != nil && *backup.Spec.MaxBackups != int32(0) {
+		nbConfigMap := generateScheduledBackupScriptsConfigMap(backup)
+		err = bm.clientSet.ConfigMap().CreateOrUpdateConfigMap(nbConfigMap)
+		if err != nil {
+			return fmt.Errorf("create config map for scheduled backup scripts err: %w", err)
+		}
+	}
+
```

Comment:
I think using ScheduledBackupScripts to running cron backup  is not a  good method,  it's so heavy.
In my view, nebula backup doesn't need to consider schedule matters, only do backup and restore job, nebulaScheduleBackup's reconciler will do these, we can use cronJob's logic in  nebulaScheduleBackup, nebulaScheduleBackup calculate next backup time by itself

Reply:
@kqzh I think there may be a misunderstanding here. The ScheduledBackupScripts do not run the cron job itself. All the cron logic for calculating the next backup time is currently taken care of by the nebulaScheduledBackup's reconciler. The script adds the logic to delete old backups that either exceed the MaxReservedTime or the MaxBackups depending on which one is set. This is not done if a user triggers a normal NebulaBackup. Will change the name of the function to generateBackupCleanupScriptsConfigMap and add a comment to make this more clear.

Reply:
Moved